lib/elixir/lib/calendar/iso.ex

```diff
@@ -337,6 +342,122 @@ defmodule Calendar.ISO do
     Integer.mod(date_to_iso_days(year, month, day) + 5, 7) + 1
   end
 
+  @doc """
+  Calculates the day of the year from the given `year`, `month`, and `day`.
+
+  It is an integer from 1 to 366.
+
+  ## Examples
+
+      iex> Calendar.ISO.day_of_year(2016, 1, 31)
+      31
+      iex> Calendar.ISO.day_of_year(-99, 2, 1)
+      32
+      iex> Calendar.ISO.day_of_year(2018, 2, 28)
+      59
+
+  """
+  @doc since: "1.8.0"
+  @spec day_of_year(year, month, day) :: day
+  @impl true
+  def day_of_year(year, month, day)
+      when is_integer(year) and is_integer(month) and is_integer(day) do
+    date_to_iso_days(year, month, day) - date_to_iso_days(year, 1, 1) + 1
```

Comment:
This implementation is a bit inefficient. I think we can "inline" the code in `date_to_iso_days` to something like this:

```elixir
true = day <= days_in_month(year, month)
days_before_month(month) + leap_day_offset(year, month) + day
```

Give or take +1/-1. :)

Reply:
Makes sense, I'll get it done.